### ISSUE TITLE
fips: ecdsa_keygen_pairwise_test does not compute message digest

### ISSUE THREAD

**Issue by xnox** [openssl/openssl#26010]
When performing `ecdsa_keygen_pairwise_test()` a digest is signed & verified, rather than a full message. This can be unapproved usage, meaning invalid pairwise test (also see CAVP FAQ GEN.2).

Note other pairwise tests do full message already (rsa does message encrypt/decrypt; and eddsa has internal apis that hash the full message).

Ideally, in ecdsa_keygen_pairwise_test() one would call something like EVP_PKEY_assign() EVP_DigestSign() EVP_DigestVerify() but these APIs are either deprecated, not available in FIPS provider. Also it is slightly inconvenient that ecdsa_keygen_pairwise_test() is called without a provider context.

I guess one could poke private structs inside eckey to get to signature methods. One implementation of this does expose some static functions to make this work https://gitlab.com/redhat/centos-stream/rpms/openssl/-/blob/c10s/0044-FIPS-140-3-keychecks.patch?ref_type=heads#L269

the KATs do the right thing, using PKEY & PKEY signature APIs but not too sure how to hook those in.

**Comment by beldmit** [comment]
I think our lab was happy with using the internal callbacks called from EVP_DigestSign_Init etc

**Comment by xnox** [comment]
> I think our lab was happy with using the internal callbacks called from EVP_DigestSign_Init etc

Yes, but those are now gone from fips provider after https://github.com/openssl/openssl/pull/25570 landed.

```
/usr/bin/ld: providers/libfips.a(libfips-lib-ec_key.o): in function `ecdsa_keygen_pairwise_test':
/home/xnox/upstream/openssl/crypto/ec/ec_key.c:1137:(.text+0x26f7): undefined reference to `EVP_PKEY_assign'
/usr/bin/ld: /home/xnox/upstream/openssl/crypto/ec/ec_key.c:1140:(.text+0x271a): undefined reference to `EVP_DigestSign'
/usr/bin/ld: /home/xnox/upstream/openssl/crypto/ec/ec_key.c:1145:(.text+0x2750): undefined reference to `EVP_DigestVerify'
```

Attempts to use EVP_DigestSign/EVP_DigestVerify are failing to link with v3.5+ codebase. I will see if adding Digest hashing alone is enough.
